apps/api_web/lib/api_web/event_stream/diff_server.ex

```diff
@@ -130,6 +132,11 @@ defmodule ApiWeb.EventStream.DiffServer do
     end
   end
 
+  @impl GenServer
+  def terminate(_reason, %{refs: refs}) do
+    for parent <- Map.keys(refs), do: send(parent, :exiting)
```

Comment:
Rather than trapping exits and manually sending a message, what about having the parents monitoring this process? Then BEAM will send it a message automatically when this process stops.

Reply:
Good idea, though wouldn't it contradict your point that the DiffServer supervisor should be before the Endpoint in the supervision tree? If that were the case, the Endpoint would be shut down before the DiffServers, so the connection processes wouldn't be around anymore to get the message that their DiffServer is shutting down.